### PR TITLE
[nodefw/drainer] add a drain flag and file

### PR DESCRIPTION
**Pull request by suiwombat**
## Description
Add a drain file to control the firewall if needed.

## Test Plan
`RUST_LOG=info cargo bpf run -- --drain-file /tmp/drain`

then rm or touch the file at /tmp/drain.  the firewall will unload or be prevented from loading if present.

https://github.com/MystenLabs/sui/assets/123987499/0a21b9f0-b9ca-47c4-a5b2-6217c350a75d

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: nre/bpf/nodefw/nodefw/src/main.rs [hunk 1]
```diff
@@ -93,17 +95,28 @@ async fn main() -> Result<(), anyhow::Error> {
 
     let listener = std::net::TcpListener::bind(nodefw::var!(
         "NODEFW_LISTEN_ON",
-        "127.0.0.1:8080".into(),
+        "127.0.0.1:8081".into(),
```
Comment (suiwombat) [inline nre/bpf/nodefw/nodefw/src/main.rs:98]: ignore, changed for testing

**Comment by suiwombat** [comment]
i'm withdrawing this pr after speaking with will.  there is a preference from mark to write on or off to the file vs simply checking for existence.

File: nre/bpf/nodefw/nodefw/src/drainer.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+use tracing::info;
+
+pub async fn watch(ctx: CancellationToken, path: &str) {
+    let mut interval = tokio::time::interval(std::time::Duration::from_secs(1));
+    interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+
+    loop {
+        tokio::select! {
+            _ = interval.tick() => {
+                if let Ok(_) = fs::metadata(path) {
```
Comment (williampsmith) [inline nre/bpf/nodefw/nodefw/src/drainer.rs:12]: OOC, why not path.exists() ?

**Review by williampsmith — APPROVED** [review]

File: nre/bpf/nodefw/nodefw/src/drainer.rs (hunk 2 above)
Comment (suiwombat) [reply nre/bpf/nodefw/nodefw/src/drainer.rs:12]: in the event i want access to the metadata info for some reason.

https://doc.rust-lang.org/std/fs/struct.Metadata.html

**Pull request merged**